Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -1309,21 +1323,28 @@ private extension ChatMessage {
 
         if let currentUser = context.currentUser {
             isSentByCurrentUser = currentUser.user.id == dto.user.id
-            currentUserReactions = Set(
-                MessageReactionDTO
-                    .loadReactions(ids: dto.ownReactions, context: context)
-                    .compactMap { try? $0.asModel() }
-            )
+            if !dto.ownReactions.isEmpty {
+                currentUserReactions = Set(
+                    MessageReactionDTO
+                        .loadReactions(ids: dto.ownReactions, context: context)
+                        .compactMap { try? $0.asModel() }
+                )
```

Comment:
Can't we simply do this?
```suggestion
                currentUserReactions = dto.ownReactions..compactMap { try? $0.asModel() }
```

Reply:
Unfortunately these are ids, not DTOs so the fetch request needs to stay for now.

Reply:
Ah I see 👍